### DISCUSSION TITLE
How to change to custom RPC

### DISCUSSION THREAD

**Discussion by alice202308** [WalletConnect/walletconnect-monorepo#3708] [Developer Support] [answered]
I want to use custom RPC instead of WalletConnect public RPC, how to change that?
<img width="713" alt="截屏2023-09-26 14 19 00" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/143504938/bdd52162-5aa6-421b-bc62-9c5beff4bbe3">

**Comment by boidushya** [comment, marked as answer]
You can modify RPC url using wagmi's [`jsonRpcProvider`](https://wagmi.sh/react/providers/jsonRpc)

**Reply by alice202308** [reply]
Thanks, but should I also change `w3mConnectors` to some other connector?

**Reply by boidushya** [reply]
Nope, w3mConnectors should work fine as is.
Here's a [MRE](https://codesandbox.io/p/sandbox/web3modal-custom-rpc-gh9rp7?file=/src/App.tsx:13,1) for future reference!
